PRASCore.jl/src/Results/DemandResponseEnergySamples.jl

```diff
@@ -0,0 +1,87 @@
+"""
+    DemandResponseEnergySamples
+
+The `DemandResponseEnergySamples` result specification reports the sample-level state
+of borrowed energy of `DemandResponses`, producing a `DemandResponseEnergySamplesResult`.
+
+A `DemandResponseEnergySamplesResult` can be indexed by demand response device name and
+a timestamp to retrieve a vector of sample-level borrowed energy states for
+the device in the given timestep.
+
+Example:
+
+```julia
+demandresponseenergy, =
+    assess(sys, SequentialMonteCarlo(samples=10), DemandResponseEnergySamples())
+
+samples = demandresponseenergy["MyDemandResponse123", ZonedDateTime(2020, 1, 1, 0, tz"UTC")]
+
+@assert samples isa Vector{Float64}
+@assert length(samples) == 10
+```
+
+Note that this result specification requires large amounts of memory for
+larger sample sizes. See [`DemandResponseEnergy`](@ref) for estimated average demand response
+borrowed energy when sample-level granularity isn't required.
+"""
+struct DemandResponseEnergySamples <: ResultSpec end
+
+struct DemandResponseEnergySamplesAccumulator <: ResultAccumulator{DemandResponseEnergySamples}
+
+    energy::Array{Float64,3}
```

Comment:
Should this be an Int or a Float64? If these are sample level results I think the shortfalls will always be integer values, no?